File: services/order-service/src/main/java/io/minishop/order/kafka/dto/InboundInventoryEvent.java
package io.minishop.order.kafka.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.time.Instant;

@JsonIgnoreProperties(ignoreUnknown = true)
public record InboundInventoryEvent(
        String type,
        Long reservationId,
        Long productId,
        Long orderId,
        Integer quantity,
        String status,
        Boolean idempotent,
        Instant occurredAt
) {}

File: services/order-service/src/main/kotlin/io/minishop/order/kafka/dto/InboundInventoryEvent.kt
package io.minishop.order.kafka.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.time.Instant

@JvmRecord
@JsonIgnoreProperties(ignoreUnknown = true)
data class InboundInventoryEvent(
    val type: String?,
    val reservationId: Long?,
    val productId: Long?,
    val orderId: Long?,
    val quantity: Int?,
    val status: String?,
    val idempotent: Boolean?,
    val occurredAt: Instant?,
)
